src/app/components/signer/signer.component.ts

```diff
@@ -165,6 +175,76 @@ export class SignerComponent implements OnInit {
       });
   }
 
+  private buildTransactionDataToSign() {
+    if (this.requestType == RequestType.PegnetTransaction || (this.requestType == "pegnet" && this.txType !== "burn")) {
+      const unixSeconds = Math.round(new Date().getTime() / 1000);
+      const entryContent = this.txType == 'transfer'
+        ? this.buildTransferEntry()
+        : this.buildConversionEntry();
+
+      return sha512(Buffer.concat([
+        Buffer.from('0'),
+        Buffer.from(unixSeconds.toString()),
+        Buffer.from('cffce0f409ebba4ed236d49d89c70e4bd1f1367d86402a3363366683265a242d', 'hex'),
+        Buffer.from(entryContent)
+      ]));
+
+    } else {
+      const amount = this.txMetadata.amount !== undefined
+        ? this.txMetadata.amount
+        : this.txMetadata.inputAmount;
+
+      const tx = Transaction
+        .builder()
+        .input(this.selectedFactomAddress, amount * Math.pow(10, 8))
+        .output('EC2BURNFCT2PEGNETooo1oooo1oooo1oooo1oooo1oooo19wthin', 0)
+        .build();
+
+      return tx['marshalBinarySig'];
+    }
+  }
+
+  private buildConversionEntry() {
+    return JSON.stringify({
+      version: 1,
+      transactions: [{
+          input: {
+            address: this.selectedFactomAddress,
+            amount: this.txMetadata.inputAmount * Math.pow(10, 8),
+            type: this.txMetadata.inputAsset
+          },
+          conversion: this.txMetadata.outputAsset,
+          metadata: {
```

Comment:
I think the metadata should be coming from the data passed in the request (the txMetadata field):
```
{
    "requestId": integer (required),
    "requestType": "pegnet",
    "requestInfo": {
        "txType": "conversion",
        "inputAddress": "FA..." (required),
        "inputAsset": string (required),
        "inputAmount": integer (required)
        "outputAsset": string (required),
        "txMetadata": JSON object (optional),
    }
}
```